lib/middleware.js

```diff
@@ -0,0 +1,75 @@
+/*!
+ * Copyright (c) 2020-2021 Digital Bazaar, Inc. All rights reserved.
+ */
+import bedrock from 'bedrock';
+import 'bedrock-express';
+import onHeaders from 'on-headers';
+
+bedrock.events.on('bedrock-express.init', app => {
+  app.use((req, res, next) => {
+    onHeaders(res, function() {
```

Comment:
Consider moving the contents of this function to the top level, use a signature like f(req, res, next), and calling it here.  Then do a private export of that function that could be imported in the test code for some pseudo testing with mock req/res/next inputs.